src/device.rs

```diff
@@ -2372,13 +2386,6 @@ impl TaskRuntime for Runtime {
                                 direct_entities.cross_ping_check.notify_failed_wg_connection(public_key)
                                     .await?;
                                 direct_entities.upgrade_sync.clear_accepted_session(public_key).await;
-
```

Comment:
Why did you removed that?

Reply:
I moved it into wg_controller so the logic around adding or removing stuff to/from SessionKeeper would be in one place. consolidation in wg_controller also does checks for downgraded peers so I moved this one there :)